modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/AbstractEventEndpoint.java

```diff
@@ -1180,27 +1180,51 @@ public Response getBulkConflicts(@FormParam("update") final String updateJson) t
           final Date end = Date.from(Instant.parse((String) scheduling.get(SCHEDULING_END_KEY)));
           final String agentId = Optional.ofNullable((String) scheduling.get(SCHEDULING_AGENT_ID_KEY))
             .orElse(event.getAgentId());
-          final List<MediaPackage> conflicting = getSchedulerService().findConflictingEvents(agentId, start, end);
-          if (conflicting != null && !conflicting.isEmpty() && !(
-            conflicting.size() == 1 && conflicting.get(0).getIdentifier().toString().equals(event.getIdentifier()))) {
-            conflicts.put(event.getIdentifier(), conflicting);
+
+          final List<JValue> currentConflicts = new ArrayList<>();
+
+          // Check for conflicts between the events themselves
+          events.values().forEach(oe -> oe.ifPresent(otherEvent -> {
+            if (otherEvent.getIdentifier().equals(event.getIdentifier())) {
+              // don't check event against itself
+              return;
+            }
+            final JSONObject otherScheduling = BulkUpdateUtil.addSchedulingDates(otherEvent, instructions.getScheduling());
+            final Date otherStart = Date.from(Instant.parse((String) otherScheduling.get(SCHEDULING_START_KEY)));
+            final Date otherEnd = Date.from(Instant.parse((String) otherScheduling.get(SCHEDULING_END_KEY)));
+            final String otherAgentId = Optional.ofNullable((String) otherScheduling.get(SCHEDULING_AGENT_ID_KEY))
+              .orElse(otherEvent.getAgentId());
+            if (!otherAgentId.equals(agentId)) {
+              // different agent -> no conflict
+              return;
+            }
+            if (!start.after(otherEnd) && !end.before(otherStart)) {
+              // conflict
+              currentConflicts.add(convertEventToConflictingObject(DateTimeSupport.toUTC(otherStart.getTime()),
+                DateTimeSupport.toUTC(otherEnd.getTime()), otherEvent.getTitle()));
+            }
+          }));
+
+          // Check for conflicts with other events from the database
+          final List<MediaPackage> conflicting = getSchedulerService().findConflictingEvents(agentId, start, end)
```

Comment:
Do we still need to look for conflicts if we have already detected an event has conflicts in the previous step?

Reply:
and maybe `.forEach` → `if (!events.keySet().contains(mp.getIdentifier().toString()))` → `currentConflicts.add` is easier to read￼

Reply:
We could skip checking for conflicts with other events from the Database in case we already found conflicts between the events to edit. That would mean, that the response won't necessarily include _all_ conflicts. It would be faster, but usability would suffer a bit. I think it's a trade-off. @staubesv any preference?

Reply:
@krinnewitz & @lkiesow No strong preference here. I would tend to always do the full conflict check as I would consider it unlikely that the scheduled events being edited conflict themselves (of course they can, but that is likely if the user accidentally enters some incorrect data and therefore not the main use case here).

Seen from that point, it is mainly the schedule conflict detection that counts and the usability would be better.

Reply:
fine with me.

Reply:
@lkiesow I tried to refactor the code to use `.forEach → if (!events.keySet().contains(mp.getIdentifier().toString())) → currentConflicts.add`

However, things became ugly due to checked exceptions. So I suggest to keep the code as currently is.